source/code/scxsystemlib/software/installedsoftwaredepend.cpp

```diff
@@ -425,212 +327,42 @@ Homepage: http://nfs.sourceforge.net/
-    \throws SCXInternalErrorException if freopen stdout fails
+    \param result : RPM query result
+    \throws SCXErrnoException if popen or pclose fails
     */
-    void InstalledSoftwareDependencies::GetRPMQueryResult(int argc, char * argv[], const wstring& tempFileName, std::vector<wstring>& result)
+    void InstalledSoftwareDependencies::GetRPMQueryResult(int argc, char * argv[], std::vector<wstring>& result)
     {
         // We need to redirect stdout to a file so we can capture the result of InvokeRPMQuery() call.
-        
-        // Flush stdout.
-        if (fflush(stdout) != 0)
-        {
-            throw SCXInternalErrorException(UnexpectedErrno(L"fflush(stdout) failed", errno), SCXSRCLOCATION);
-        }
-        // Make a copy of original stdout destination.
-        int oldStdOutFD = dup(1);
-        if (oldStdOutFD == -1)
-        {
-            throw SCXInternalErrorException(UnexpectedErrno(L"dup(1) failed", errno), SCXSRCLOCATION);
-        }
-        // Open new stdout destination.
-        int fileFD = open(SCXCoreLib::StrToUTF8(tempFileName).c_str(), O_CREAT | O_RDWR | O_TRUNC, S_IRUSR | S_IWUSR);
-        if (fileFD == -1)
-        {
-            throw SCXInternalErrorException(UnexpectedErrno(L"open(" + tempFileName + L") failed", errno), SCXSRCLOCATION);
-        }
-        // Redirect stdout to the new destination.
-        if (dup2(fileFD, 1) == -1)
-        {
-            throw SCXInternalErrorException(UnexpectedErrno(L"dup2(fileFD, 1) failed", errno), SCXSRCLOCATION);
-        }
-
-        InvokeRPMQuery(argc, argv);
-
-        // Flush new stdout.
-        if (fflush(stdout) != 0)
-        {
-            throw SCXInternalErrorException(UnexpectedErrno(L"fflush(stdout) failed", errno), SCXSRCLOCATION);
-        }
-        // Restore stdout to the original destination.
-        if (dup2(oldStdOutFD, 1) == -1)
-        {
-            throw SCXInternalErrorException(UnexpectedErrno(L"dup2(oldStdOutFD, 1) failed", errno), SCXSRCLOCATION);
-        }
-        // Close new stdout destination, not needed anymore.
-        if (close(fileFD) != 0)
-        {
-            throw SCXInternalErrorException(UnexpectedErrno(L"close(fileFD) failed", errno), SCXSRCLOCATION);
-        }
-        // Close copy of the original stdout destination.
-        if (close(oldStdOutFD) != 0)
-        {
-            throw SCXInternalErrorException(UnexpectedErrno(L"close(oldStdOutFD) failed", errno), SCXSRCLOCATION);
-        }
-
-        SCXCoreLib::SCXHandle<std::wfstream> fs(SCXCoreLib::SCXFile::OpenWFstream(tempFileName, std::ios::in));
-        fs.SetOwner();
-        SCXStream::NLFs nlfs;
-        SCXCoreLib::SCXStream::ReadAllLines(*fs, result, nlfs);
-        fs->close();
+	std::wstring rpmPath = L"/bin/rpm";
+	std::wstringstream commandToRun;
+
+	if (!SCXFile::Exists(SCXFilePath(rpmPath)))
+	{
+	    SCX_LOGINFO(m_log, L"No rpm executable at /bin/rpm, therefore skipping rpm package enumeration.");
```

Comment:
Do we want to log this every single time? Or do we want to log once and then don't log again?